spring-cloud-aws-autoconfigure/src/main/java/io/awspring/cloud/autoconfigure/core/StsProperties.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2013-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.awspring.cloud.autoconfigure.core;
+
+import java.nio.file.Path;
+import org.springframework.lang.Nullable;
+
+/**
+ * Properties related to AWS Sts Credentials. It the properties are not configured, it will default to the EKS values
+ * from: <a href="https://docs.aws.amazon.com/eks/latest/userguide/pod-configuration.html">
+ *
+ * @author Eduan Bekker
+ * @since 3.0.0
+ */
+public class StsProperties {
+
+	/**
+	 * Activate {@link software.amazon.awssdk.services.sts.auth.StsWebIdentityTokenFileCredentialsProvider}
+	 * AutoConfiguration for {@link software.amazon.awssdk.crt.auth.credentials.CredentialsProvider}.
+	 */
+	private boolean enabled = false;
```

Comment:
I would like to keep this since I am not a fan of autoconfiguring STS based only on if the class is on the classpath. There are multiple ways you can do STS and we should not force it.
This also helps if you want locally not to use STS, but once deployed to use it.

Reply:
I agree this should be configurable. I wonder if we should make the default `true` because you already need to add the dependency and it is common for Spring Boot to auto-configure things based on dependencies. Additionally, you need to set the role and have a credentials file so I don't think it will happen accidentally locally. I also like the idea of zero-config authentication in EKS. 

That said, a false default is also fine by me.

Reply:
I agree with all of this, but imagine someone using STS for one client and (accessKey, secretKey) for all other clients. This way he can just disable STS autoconfiguration.
Was thinking about this as well, I know it is a weird use case but I can see it happening.